x/gamm/keeper/taker_fee.go

```diff
@@ -17,57 +15,17 @@ func (k Keeper) chargeTakerFee(
 	ctx sdk.Context,
 	takerFeeCoin sdk.Coin,
 	sender sdk.AccAddress,
-	route poolmanagertypes.SwapAmountInRoute,
```

Comment:
`chargeTakerFee` used to swap if it's unknown denom.
as we don't expect pools with non-dym, we don't need this logic.
And if we would like it in the future, it needs to be part of the `x/txfees` logic anyway

Reply:
I don't think it's correct we don't expect pools with non-dym.
we may have usdc pools.